### PR TITLE
 JAMES-3112 Distinguish Domain Alias from Domain Mapping RRTs 

### PR DESCRIPTION
**Pull request by chibenwa**
Domain mapping is intended for technical purposes, and should not be
taken into account for user allowed from.

Domain aliases are user exposed, and should be taken into account for
user eallowed from.

### CODE REVIEW

**Comment by chibenwa** [comment]
Needs upgrade instructions & changelog

**Comment by chibenwa** [comment]
`MemorySpamAssassinContractTest.imapMovesToSpamMailboxShouldBeConsideredAsSpam » ConditionTimeout`

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
build crashed at the start of the pod... 

test this please

**Comment by Arsnael** [comment]
```
[1b73e2a392caf6f00954b77ee3fe33d24f695740] [ERROR]   SolveMailboxInconsistenciesServiceTest.fixMailboxInconsistenciesShouldReturnPartialWhenDAOMisMatchOnId:157 
[1b73e2a392caf6f00954b77ee3fe33d24f695740] Expecting:
[1b73e2a392caf6f00954b77ee3fe33d24f695740]  <COMPLETED>
[1b73e2a392caf6f00954b77ee3fe33d24f695740] to be equal to:
[1b73e2a392caf6f00954b77ee3fe33d24f695740]  <PARTIAL>
[1b73e2a392caf6f00954b77ee3fe33d24f695740] but was not.
```
doesn't sound related... test this please

File: server/data/data-api/src/test/java/org/apache/james/rrt/lib/CanSendFromContract.java [hunk 1]
```diff
@@ -134,13 +134,22 @@ default void userCanSendFromShouldBeTrueWhenSenderIsAnAliasInAnotherDomainOfAnAl
...
         redirectDomain(OTHER_DOMAIN).to(DOMAIN);
 
         assertThat(canSendFrom().userCanSendFrom(USER, fromUser)).isTrue();
     }
 
+    @Test
+    default void userCanSendFromShouldBeFalseWhenDomainMapping() throws Exception {
+        Username fromUser = USER.withOtherDomain(OTHER_DOMAIN);
+
+        addDomainMapping(OTHER_DOMAIN, DOMAIN);
```
Comment (blackheaven) [inline server/data/data-api/src/test/java/org/apache/james/rrt/lib/CanSendFromContract.java]: Use `redirect(..).to(..)` syntax

Comment (chibenwa) [reply server/data/data-api/src/test/java/org/apache/james/rrt/lib/CanSendFromContract.java]: Would you agree with `redirect(...).to(...).asMapping()` & `redirect(...).to(...).asAlias()`?

Comment (blackheaven) [reply server/data/data-api/src/test/java/org/apache/james/rrt/lib/CanSendFromContract.java]: sounds nice

**Comment by Arsnael** [comment]
Tests are :green_apple: , deployment test failed...

test this please

File: upgrade-instructions.md [hunk 2]
```diff
@@ -29,6 +29,24 @@ Change list:
...
  - [UidValidity and JPA or Cassandra](#uid-validity-and-jpa-or-cassandra)
+ - [Differentiation between domain alias and domain mapping](#differentiation-between-domain-alias-and-domain-mapping)
+ 
+### Differentiation between domain alias and domain mapping
+
+Date 10/03/2020
+
+SHA-1 XXX
+
+JIRA: https://issues.apache.org/jira/browse/JAMES-XXXX
```
Comment (Arsnael) [inline upgrade-instructions.md]: JAMES-3112?

File: upgrade-instructions.md [hunk 3]
```diff
@@ -29,6 +29,24 @@ Change list:
...
+
+Concerned products: JPA mailbox backend, Cassandra mailbox backend
+
+Non-Maildir backends could generate '0' uid validity with a low probability, which is invalid regarding RFC-3501. 
+We changed the generation mechanism to use valid UidValidity for newly created mailboxes. Regarding persisted mailboxes,
+we regenerate invalid UidValidity upon reads.
+
+While this sanitizing is transparent to the end user and the admin, it might lead to rare IMAP client full mailbox
+ resynchronisation. (one chance out of two billions).
```
Comment (Arsnael) [inline upgrade-instructions.md]: Didn't you mix up here? It should be able domain alias and domain mapping diff, not UidValidity?

**Comment by Arsnael** [comment]
```
[e42e303f3736faac31d9b1def88dd0adbf4cee02] [ERROR] Errors: 
[e42e303f3736faac31d9b1def88dd0adbf4cee02] [ERROR]   MemorySpamAssassinContractTest.expungingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout
```
test this please

File: upgrade-instructions.md [hunk 4]
```diff
@@ -37,16 +37,20 @@ Date 10/03/2020
...
+[endpoint](https://github.com/apache/james-project/blob/master/src/site/markdown/server/manage-webadmin.md#get-the-list-of-aliases-for-a-domain))
+to distinguish them from domain mappings (exposed by this 
+[endpoint](https://github.com/apache/james-project/blob/master/src/site/markdown/server/manage-webadmin.md#creating-address-domain-aliases)).
 
-While this sanitizing is transparent to the end user and the admin, it might lead to rare IMAP client full mailbox
- resynchronisation. (one chance out of two billions).
+Read [this page](https://james.apache.org/server/config-recipientrewritetable.html) to understand the difference between 
+Domain Alias and Domain Mapping.
+
+As a consequence, existing value returned by the domain alias endpoint (before this fix this is domain mapping) will be
```
Comment (Arsnael) [inline upgrade-instructions.md]: s/value/values

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**